From 35ee8e451b457b2e85f7c6a5c3d50cbcb343fe01 Mon Sep 17 00:00:00 2001
From: jizhenlo <zhenlong.z.ji@intel.com>
Date: Thu, 9 Nov 2023 22:51:12 +0800
Subject: [PATCH] Remove the existence check of mkfs.exfat and fsck.exfat

mkfs.exfat and fsck.exfat is introduced in android 13. For
android 12, U disk with exfat fs still can be accessed with
exfat fs enabled in kernel.

Tracked-On: OAM-113303
Signed-off-by: jizhenlo <zhenlong.z.ji@intel.com>
---
 fs/Exfat.cpp | 37 +++++--------------------------------
 1 file changed, 5 insertions(+), 32 deletions(-)

diff --git a/fs/Exfat.cpp b/fs/Exfat.cpp
index 7782dd3b..bef66fba 100644
--- a/fs/Exfat.cpp
+++ b/fs/Exfat.cpp
@@ -34,25 +34,11 @@ static const char* kMkfsPath = "/system/bin/mkfs.exfat";
 static const char* kFsckPath = "/system/bin/fsck.exfat";
 
 bool IsSupported() {
-    return access(kMkfsPath, X_OK) == 0 && access(kFsckPath, X_OK) == 0 &&
-           IsFilesystemSupported("exfat");
+    return IsFilesystemSupported("exfat");
 }
 
 status_t Check(const std::string& source) {
-    std::vector<std::string> cmd;
-    cmd.push_back(kFsckPath);
-    cmd.push_back("-y");
-    cmd.push_back(source);
-
-    int rc = ForkExecvp(cmd, nullptr, sFsckUntrustedContext);
-    if (rc == 0) {
-        LOG(INFO) << "Check OK";
-        return 0;
-    } else {
-        LOG(ERROR) << "Check failed (code " << rc << ")";
-        errno = EIO;
-        return -1;
-    }
+	return 0;
 }
 
 status_t Mount(const std::string& source, const std::string& target, int ownerUid, int ownerGid,
@@ -75,22 +61,9 @@ status_t Mount(const std::string& source, const std::string& target, int ownerUi
 }
 
 status_t Format(const std::string& source) {
-    std::vector<std::string> cmd;
-    cmd.push_back(kMkfsPath);
-    cmd.push_back("-n");
-    cmd.push_back("android");
-    cmd.push_back(source);
-
-    int rc = ForkExecvp(cmd);
-    if (rc == 0) {
-        LOG(INFO) << "Format OK";
-        return 0;
-    } else {
-        LOG(ERROR) << "Format failed (code " << rc << ")";
-        errno = EIO;
-        return -1;
-    }
-    return 0;
+	LOG(ERROR) << "Format failed";
+	errno = EIO;
+	return -1;
 }
 
 }  // namespace exfat
-- 
2.25.1

